File: R/easy_glmnet.R
# The following information allows one to reproduce the results
# of the analysis in R.
# NOTE: easyml may evolve which is why a commit ID is provided.
# Repository: https://github.com/CCS-Lab/easyml
# Commit ID: dac38d697ab5adf2faaaab68d6aeb5bd0d729a94
# URL: https://github.com/CCS-Lab/easyml/commit/dac38d697ab5adf2faaaab68d6aeb5bd0d729a94
library(easyml)

# Load data
data("cocaine_dependence", package = "easyml")

# Analyze data
results <- easy_glmnet(cocaine_dependence, "diagnosis",
                       family = "binomial", preprocess = preprocess_scale, 
                       exclude_variables = c("subject"), categorical_variables = c("male"), 
                       n_samples = 10, n_divisions = 100, n_iterations = 10
                       random_state = 12345, model_args = list(alpha = 1.0))

results$plot_coefficients
results$plot_model_performance_train
results$plot_model_performance_test
results$plot_roc_single_train_test_split_train
results$plot_roc_single_train_test_split_test

File: Python/easy_glmnet.py
"""
The following information allows one to reproduce the results
of the analysis in Python.
NOTE: easyml may evolve which is why a commit ID is provided.
Repository: https://github.com/CCS-Lab/easyml
Commit ID: 4f21118ae2b531a2fa3993826d8084455c58398f
URL: https://github.com/CCS-Lab/easyml/tree/4f21118ae2b531a2fa3993826d8084455c58398f
"""
import matplotlib.pyplot as plt
import os
import pandas as pd

from easyml.glmnet import EasyGlmnet


# Set matplotlib settings
plt.style.use('ggplot')

# Load data
directory = '../data/'
cocaine_dependence = pd.read_table(os.path.join(directory, 'cocaine_dependence.txt'))

# Analyze data
output = EasyGlmnet(cocaine_dependence, 'DIAGNOSIS',
                    family='binomial',
                    exclude_variables=['subject'],
                    categorical_variables=['Male'],
                    random_state=12345, progress_bar=True, n_core=1,
                    model_args={'alpha': 1, 'n_lambda': 200})
